server/webapp/WEB-INF/rails/webpack/models/mixins/new_validatable_mixin.ts

```diff
@@ -111,22 +111,40 @@ class MaxLengthValidator extends Validator {
   }
 }
 
-class MutualExclusivityValidator extends Validator {
+class MutualExclusivityValidator<T = {}> extends Validator {
   private attrTwo: string;
   constructor(attrTwo: string, options: ValidatorOptions = {}) {
     super(options);
     this.attrTwo = attrTwo;
   }
 
   protected doValidate(entity: any, attrName: string): void {
-    if ((entity[attrName]() === undefined) || (entity[this.attrTwo]() === undefined)) {
+    if (this.isEmpty(entity[attrName]()) && this.isEmpty(entity[this.attrTwo]())) {
+      entity.errors().add(attrName, this.options.message || ErrorMessages.mutuallyExclusive(attrName, this.attrTwo));
       return;
     }
 
-    if (!(entity[attrName]().length === 0) && !(entity[this.attrTwo]().length === 0)) {
+    if (!this.isEmpty(entity[attrName]()) && !this.isEmpty(entity[this.attrTwo]())) {
       entity.errors().add(attrName, this.options.message || ErrorMessages.mutuallyExclusive(attrName, this.attrTwo));
     }
   }
+
+  private isEmpty(i: Iterable<T>): boolean {
```

Comment:
I might just pull this out as a util method. The validator shouldn't need a type param after that.